src/main/kotlin/io/beatmaps/ws/server.kt

```diff
@@ -72,6 +72,18 @@ fun Application.ws() {
                 queueDeclare("ws.voteStream", true, false, false, genericQueueConfig)
                 queueBind("ws.voteStream", "beatmaps", "voteupdate.*")
 
+                queueDeclare("ws.reviewStream.created", true, false, false, genericQueueConfig)
+                queueBind("ws.reviewStream.created", "beatmaps", "reviews.*.created")
+
+                queueDeclare("ws.reviewStream.updated", true, false, false, genericQueueConfig)
+                queueBind("ws.reviewStream.updated", "beatmaps", "reviews.*.updated")
+
+                queueDeclare("ws.reviewStream.deleted", true, false, false, genericQueueConfig)
+                queueBind("ws.reviewStream.deleted", "beatmaps", "reviews.*.deleted")
+
+                queueDeclare("ws.reviewStream.curated", true, false, false, genericQueueConfig)
+                queueBind("ws.reviewStream.curated", "beatmaps", "reviews.*.curated")
```

Comment:
If you want consistent message order you probably don't want 4 queues.
Just bind `reviews.*`

Yes, you'll then have to unpick the type of message later (maps send extra data in the message itself but it would also be possible to get the original routingKey from `message.envelope.routingKey`)

Reply:
I currently don't have access to the `message.envelope` object and copying the code from consumeAck to gain access, seems kinda dirty. But using `message.envelope.routingkey` seems preferable. I'm really not sure how to utilize it though without making a mess.

Adding a third parameter to `consumeAck` has crossed my mind, but that doesn't seem great either.